File: src/main/java/tech/libeufin/messages/HEVResponse.java
package tech.libeufin.messages;

import javax.xml.bind.JAXBElement;


public class HEVResponse {
    HEVResponseDataType value;

    public HEVResponse(String returnCode, String reportText){
        SystemReturnCodeType srt = new SystemReturnCodeType();
        srt.setReturnCode(returnCode);
        srt.setReportText(reportText);
        this.value = new HEVResponseDataType();
        this.value.setSystemReturnCode(srt);
    }

    public JAXBElement<HEVResponseDataType> makeHEVResponse(){
        ObjectFactory of = new ObjectFactory();
        return of.createEbicsHEVResponse(this.value);
    }
}

File: src/main/kotlin/tech/libeufin/messages/HEVResponse.kt
package tech.libeufin.messages

import javax.xml.bind.JAXBElement


class HEVResponse(
    returnCode: String,
    reportText: String,
    protocolAndVersion: Array<ProtocolAndVersion>?) {

    constructor(
        returnCode: String,
        reportText: String
    ) : this(returnCode, reportText, null)

    private val value: HEVResponseDataType = {
        val srt = SystemReturnCodeType()
        srt.setReturnCode(returnCode);
        srt.setReportText(reportText);
        val value = HEVResponseDataType();
        value.setSystemReturnCode(srt);

        protocolAndVersion?.forEach {
            val entry = HEVResponseDataType.VersionNumber()
            entry.setProtocolVersion(it.protocol)
            entry.setValue(it.version)
            value.getVersionNumber().add(entry)
        }

        value
    }()

    fun makeHEVResponse(): JAXBElement<HEVResponseDataType> {
        val of = ObjectFactory()
        return of.createEbicsHEVResponse(value)
    }
}
